protos/guard/guard.proto

```diff
@@ -240,6 +241,29 @@ message CheckFileStoreMetaRequest {
   bytes signature = 5; //requester signed
 }
 
+message ListAllContractsRequest {
+  string host_pid = 1;
+  string requester_pid = 2;
+  google.protobuf.Timestamp request_time = 3 [
```

Comment:
will this return contracts that has updated time > request_time? we don't want to return ALL because of increasing history

also consider using a filter to get status - active vs terminated vs all

Reply:
so you suggest add some filter such like contract status and contract rental period?

Reply:
yes